packages/cipherstash-proxy/src/config/log.rs

```diff
@@ -0,0 +1,211 @@
+use std::{fmt::Display, io::IsTerminal};
+
+use clap::ValueEnum;
+use serde::Deserialize;
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct LogConfig {
+    #[serde(default = "LogConfig::default_ansi_enabled")]
+    pub ansi_enabled: bool,
+
+    #[serde(default = "LogConfig::default_log_format")]
+    pub format: LogFormat,
+
+    #[serde(default = "LogConfig::default_log_output")]
+    pub output: LogOutput,
+
+    #[serde(default = "LogConfig::default_log_level")]
+    pub level: LogLevel,
+
+    #[serde(default = "LogConfig::default_log_level")]
+    pub development_level: LogLevel,
+
+    #[serde(default = "LogConfig::default_log_level")]
+    pub authentication_level: LogLevel,
+
+    #[serde(default = "LogConfig::default_log_level")]
+    pub config_level: LogLevel,
+
+    #[serde(default = "LogConfig::default_log_level")]
+    pub context_level: LogLevel,
+
+    #[serde(default = "LogConfig::default_log_level")]
+    pub encoding_level: LogLevel,
+
+    #[serde(default = "LogConfig::default_log_level")]
+    pub encrypt_level: LogLevel,
+
+    #[serde(default = "LogConfig::default_log_level")]
+    pub encrypt_config_level: LogLevel,
+
+    #[serde(default = "LogConfig::default_log_level")]
+    pub keyset_level: LogLevel,
+
+    #[serde(default = "LogConfig::default_log_level")]
+    pub migrate_level: LogLevel,
+
+    #[serde(default = "LogConfig::default_log_level")]
+    pub protocol_level: LogLevel,
+
+    #[serde(default = "LogConfig::default_log_level")]
+    pub mapper_level: LogLevel,
+
+    #[serde(default = "LogConfig::default_log_level")]
+    pub schema_level: LogLevel,
+}
+
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, ValueEnum)]
+#[serde(rename_all = "lowercase")]
+pub enum LogFormat {
+    // Serde does not seem to have a case insensitive option. alias is clunky, but better than custom de/serialisers
+    #[serde(alias = "Pretty", alias = "pretty", alias = "PRETTY")]
```

Comment:
TIL about alias.

Reply:
@coderdan To be fair, it's a pretty ugly workaround. Alternative was writing it from scratch, but just seemed not worth the effort to handle typos in log config options.